src/pages/{uri}.js

```diff
@@ -0,0 +1,17 @@
+import { createPagesFromData, graphql } from "gatsby"
+
+// Renders the component from the node
+function Component({ data }) {
+  return <data.Component {...data} />
+}
+
+export default createPagesFromData(Component, `allWpContentNodes`)
+
+export const query = graphql`
+  query Node($id: String!) {
+    WpContentNodes(id: $id) {
+      ...someFields
+      Component # this is based on the hypothetical work from Mikhail
```

Comment:
Probably more a question for Mikhail, but the `Coponent` of any given node could be different depending on how it's queried. So a component should probably be edge data, not a field of the node itself. 

For example, a Post as displayed on an Archive Page of all posts would have a different Component to render it than the same Post displayed individually at its own URL and another different Component if displayed as part of a Related Posts query in the sidebar, for example. 

If we were going to use GraphQL to return components, I think we'll need to really think through how the context of a node would affect it's rendering. 

But. . .if Components were exposed on the edge, this could _maybe_ work for us 🤔

Reply:
@jasonbahl that's true but I think in this case it'd be a little different as we'd just have to be more explicit about it. 

The components would be more like this:
```gql
query Node($id: String!) {
    WpContentNodes(id: $id) {
      ...someFields
      SingleComponent
      ArchiveComponent
```
When querying for a post in a list on an archive page we wouldn't need the component/template for it since it's being displayed in an archive template already

Reply:
And the reason is that archive templates aren't based on single nodes but are based on information about all single nodes of 1 type

Reply:
Ok 🤦 so my example above is wrong then :p because you would never query for the archive template on a single node haha.

Reply:
So 1 node would only ever resolve to a single template since we can only create 1 page at each uri. Archive template would be completely separate logic

Reply:
Yeah.. I'm thinking that the template would be decided by your logic in the onCreateNode. Idk exactly how that all works so I'm just guessing